### PR TITLE
Gaussian peak value helpers

### PR DESCRIPTION
**Pull request by vidartf**
Added convenience helper property to Gaussian which sets and gets the function value
at the peak. It is simply parameter 'A' scaled by sigma*sqrt(2*Pi).

### CODE REVIEW

**Comment by vidartf** [comment]
@francisco-dlp : This one now has (have had for quite a while actually) "an extra gaussian component written as a function of fwhm and height instead of sigma and area" as proposed above. Opinions?

Very useful as you can fix the max value (height), while fitting the sigma and/or centre parameters.

**Comment by vidartf** [comment]
PS: I think this could help improve the new EDSModel (#786).

File: hyperspy/_components/gaussian2.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+import numpy as np
+
+from hyperspy.component import Component
+
+sqrt2pi = math.sqrt(2 * math.pi)
+sigma2fwhm = 2 * math.sqrt(2 * math.log(2))
+
+
+class Gaussian2(Component):
```
Comment (francisco-dlp) [inline hyperspy/_components/gaussian2.py]: What about calling this `GaussianHF` instead—I think that it is a little bit more telling than `Gaussian2`

File: hyperspy/_components/gaussian2.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    | Parameter  | Attribute |
+    +------------+-----------+
+    +------------+-----------+
+    |     a      |  height   |
+    +------------+-----------+
+    |     b      |  centre   |
+    +------------+-----------+
+    |     c      |   fwhm    |
+    +------------+-----------+
+
```
Comment (francisco-dlp) [inline hyperspy/_components/gaussian2.py]: You could add a See Also section to point to `Gaussian`

Comment (francisco-dlp) [inline hyperspy/_components/gaussian2.py]: It would be good to document the `A` and `sigma` attributes here.

**Comment by francisco-dlp** [comment]
* Given that we now have `Expression`, shouldn't this be implemented using `Expression` instead? Actually [here you have it](http://hyperspy.org/hyperspy-doc/current/user_guide/model.html#adding-components-to-the-model). It should be possible to add the parameter estimation on top of the `Expression` component.
* As everything else, this needs unittests and a small mention in the User Guide.

**Comment by vidartf** [comment]
I could replace it with an `Expression`, but what are the advantages? Since Expression was designed for run-time use, it also doesn't work very well for inheritance, e.g. the class doc is overwritten...

**Comment by francisco-dlp** [comment]
It runs faster when using `numexpr`, the code is far more compact and
likelihood of a bug (example of typical bugs are wrong grads) greatly
decreases. The main disadvantage is no sympy -> no component, Currently
HyperSpy can run without sympy, but sympy is in setup.py.

2015-12-15 15:26 GMT+00:00 Vidar Tonaas Fauske <notifications@github.com>:

**Comment by vidartf** [comment]
Does `Expression` support binning?

**Comment by francisco-dlp** [comment]
"binning" should be provided by `Model`. Only `estimate_parameters` should
require a separate case for binning.

2015-12-15 18:32 GMT+00:00 Vidar Tonaas Fauske <notifications@github.com>:

**Comment by vidartf** [comment]
Alright, reasonably happy with test coverage, and code should be working (?).

**Comment by francisco-dlp** [comment]
Cool, nicely done. Some little comments:

`estimate_parameters` is almost identical to the one in `Gaussian`. Wouldn't it be better to turn the method into a function so that both components can use the same code?

I have some concerns about the tests:

* `integral_as_signal` and `A` and `sigma` are not tested. 
* The tests test too much. `Expression` is already well tested, so, actually, there is not much left to test. I think that just testing that the evaluating the function gives the right result should be enough. The another simple test for `estimate_parameters` and that should be all. Using the component to fit a model is actually testing `fit` and `multifit`, but there is no need to do that here.

**Comment by francisco-dlp** [comment]
Oh, and there is not mention to this in the User Guide. Just listing it would be enough.

**Comment by vidartf** [comment]
I think I've added the things asked for. Not that due to the rather low accuracy of the `estimate_parameters` function, `nt.assert_almost_equal` could not be used very much.

**Comment by francisco-dlp** [comment]
I have sent you a PR that mainly aims at simplifying the test. `estimate_parameters` is only accurate when a big fraction of the Gaussian is in the estimation region.

**Pull request merged**